integration-tests/src/cross_chain_transfer.rs

```diff
@@ -487,3 +487,202 @@ fn assets_should_be_trapped_when_assets_are_unknown() {
 		assert_eq!(basilisk_runtime::PolkadotXcm::asset_trap(hash), 1);
 	});
 }
+
+#[test]
+fn transfer_from_hydra_and_back() {
+	TestNet::reset();
+
+	Basilisk::execute_with(|| {
+		assert_ok!(basilisk_runtime::AssetRegistry::set_location(
+			basilisk_runtime::Origin::root(),
+			1,
+			basilisk_runtime::AssetLocation(MultiLocation::new(1, X2(Parachain(3000), GeneralIndex(0))))
+		));
+	});
+
+	Hydra::execute_with(|| {
+		assert_ok!(basilisk_runtime::XTokens::transfer(
+			basilisk_runtime::Origin::signed(ALICE.into()),
+			0,
+			30 * UNITS,
+			Box::new(
+				MultiLocation::new(
+					1,
+					X2(
+						Junction::Parachain(2000),
+						Junction::AccountId32 {
+							id: BOB,
+							network: NetworkId::Any,
+						}
+					)
+				)
+				.into()
+			),
+			399_600_000_000
+		));
+		assert_eq!(
+			basilisk_runtime::Balances::free_balance(&AccountId::from(ALICE)),
+			200 * UNITS - 30 * UNITS
+		);
+	});
+
+	Basilisk::execute_with(|| {
+		assert_eq!(
+			basilisk_runtime::Tokens::free_balance(1, &AccountId::from(BOB)),
+			1_029_999_989_814_815
+		);
+		assert_eq!(
+			basilisk_runtime::Tokens::free_balance(1, &basilisk_runtime::Treasury::account_id()),
+			10_185_185 // fees should go to treasury
+		);
+
+		//transfer back
+		assert_ok!(basilisk_runtime::MultiTransactionPayment::set_currency(
+			basilisk_runtime::Origin::signed(BOB.into()),
+			1,
+		));
+
+		assert_ok!(basilisk_runtime::XTokens::transfer(
+			basilisk_runtime::Origin::signed(BOB.into()),
+			0,
+			30 * UNITS,
+			Box::new(
+				MultiLocation::new(
+					1,
+					X2(
+						Junction::Parachain(3000),
+						Junction::AccountId32 {
+							id: ALICE,
+							network: NetworkId::Any,
+						}
+					)
+				)
+				.into()
+			),
+			399_600_000_000
+		));
+		assert_eq!(
+			basilisk_runtime::Balances::free_balance(&AccountId::from(BOB)),
+			1000 * UNITS - 30 * UNITS
+		);
+		assert_eq!(
+			basilisk_runtime::Tokens::free_balance(1, &basilisk_runtime::Treasury::account_id()),
+			35_990_141 // fees should go to treasury
+		);
```

Comment:
why not check that they arrive back on the Hydra chain?

Reply:
i am not sure anymore why  this test was there at all as part of this PR. i removed it. this scenarios are already covered by other testcases.